error/AppErrors.ts

```diff
@@ -1,42 +1,53 @@
 const i18n = require('../loc');
 
-export class AppError extends Error {
-  constructor(message?: string) {
-    super(message);
-  }
-}
-
-export class DoubleSpentFundsError extends AppError {
+// TODO: Refactor this - the default message should be always in English
+// Also, I dont' see a point of exporting everything in index.ts under the same namespace
+export class DoubleSpentFundsError extends Error {
   constructor(message = i18n.send.error.doubleSpentFunds) {
     super(message);
+
+    Object.setPrototypeOf(this, DoubleSpentFundsError.prototype);
```

Comment:
I don't exactly get an idea what is the point of doing it?

Reply:
1. It's recommended way but the official documentation
2. It makes that` DoubleSpentFundsError` is an instance of `Error`. Without it:

Reply:
can you send the link?

Reply:
The link is in the PR description

Reply:
there is a warning https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf on a Mozilla, maybe it's worth checking the alternative solution, as this is from TS v2 as I see

Reply:
To be honest, I did it because it was in the documentation. I have never spent much time reading about prototypes.  You're more experienced so if you're saying to remove it - I will. I simply don't have a strong opinion here

Reply:
<img width="1075" alt="Zrzut ekranu 2021-02-3 o 15 15 20" src="https://user-images.githubusercontent.com/67273035/106759142-ac4c1f80-6632-11eb-9f8f-b291bbbfae7a.png">
from what I understand the problem occurs for compiling to es5 and lower ecma versions. Our target is set to ES2019, so there shouldn't be a problem

Reply:
Cool, good to know. Thanks for making research. I'll remove it then